website/docs/guides/migrating-from-deprecated-resources.html.markdown

```diff
@@ -0,0 +1,156 @@
+---
+layout: "azurerm"
+page_title: "Azure Provider: Migrating from Deprecated Resources Guide"
+description: |-
+This page documents how to migrate from deprecated resources in the Azure Provider to their replacements.
+
+---
+
+# Azure Provider: Migrating from Deprecated Resources Guide
+
+This guide shows how to migrate from a resource which has been deprecated or renamed to its replacement.
+
+It's possible to migrate between the resources by updating your Terraform Configuration, removing the old state, and the importing the new resource in config.
+
+In this guide, we'll assume we're migrating from the `azurerm_app_service` resource to the new `azurerm_linux_web_app` resource, but this should also be applicable for resource that have simply been renamed where you can simply change the resource type name in your config.
+
+Assuming we have the following Terraform Configuration:
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  # ...
+}
+
+resource "azurerm_app_service_plan" "example" {
+  name                = "Example App Service Plan"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+  kind                = "Linux"
+  reserved            = true
+
+  sku {
+    tier = "Standard"
+    size = "S1"
+  }
+}
+
+resource "azurerm_app_service" "example" {
+  name                = "Example App Service"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+  app_service_plan_id = azurerm_app_service_plan.main.id
+
+  site_config {
+    dotnet_framework_version = "v4.0"
+    remote_debugging_enabled = true
+    remote_debugging_version = "VS2019"
+  }
+}
+```
+
+We can update the Terraform Configuration to use the new resource by updating the resources to the new `azurerm_service_plan` and `azurerm_linux_web_app` schema:
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  # ...
+}
+
+resource "azurerm_service_plan" "example" {
+  name                = "Example App Service Plan"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = "West Europe"
+  os_type             = "Linux"
+  sku_name            = "P1V2"
```

Comment:
Should probably match this up to the source plan SKU from the resource being migrated from above to avoid confusion?
```suggestion
  sku_name            = "B1"
```